engines/e_devcrypto.c

```diff
@@ -1172,17 +1219,29 @@ static int devcrypto_ctrl(ENGINE *e, int cmd, long i, void *p, void (*f) (void))
  */
 static int open_devcrypto(void)
 {
+    int fd;
+
     if (cfd >= 0)
         return 1;
 
-    if ((cfd = open("/dev/crypto", O_RDWR, 0)) < 0) {
+    if ((fd = open("/dev/crypto", O_RDWR, 0)) < 0) {
 #ifndef ENGINE_DEVCRYPTO_DEBUG
         if (errno != ENOENT)
```

Comment:
It would have been handy during another round of testing I did today where I had forgotten to load the cryptodev.ko module on FreeBSD (which implements the `/dev/crypto` character device) if this error had not been silent.  However, this would be too noisy normally.  What I would like is that if a user explicitly asks for the engine, e.g. via a `-engine devcrypto` command line argument to, say, `openssl speed`, that this would print an error for `ENOENT`, but that if openssl is just trying all the engines to see which ones work, this error would stay silent.  Not sure if there is a way to detect that difference inside an engine itself to tailor the behavior.  If not, the current behavior is probably the most correct one.

Reply:
> Not sure if there is a way to detect that difference inside an engine itself to tailor the behavior.

The engines are completely agnostic to this, so it's not possible.